natural_language_processing/word_embeddings/word_embeddings.py

```diff
@@ -54,8 +56,7 @@ def similarity(self, w1: str, w2: str) -> float:
         w1, w2 = w1.lower(), w2.lower()
         error_msg = 'both words must be in the vocabulary'
         assert self.is_in_vocab(w1) and self.is_in_vocab(w2), error_msg
```

Comment:
If the assertions are for tests, use the doctest format (see [here](https://docs.python.org/3/library/doctest.html))

Reply:
These assertions are for runtime checks...  If the user gives me garbage then I should complain.  These garbage values should also be tested in the doctests.